src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -432,14 +432,77 @@ p2ptest_chunk(xclDeviceHandle handle, char *boptr, uint64_t dev_addr, uint64_t s
   return true;
 }
 
+static bool
+p2ptest_chunk_no_dma(xclDeviceHandle handle, xclBufferHandle bop2p, size_t bo_size, int bank)
+{
+
+  xclBufferHandle boh = xclAllocBO(handle, bo_size, 0, XCL_BO_FLAGS_HOST_ONLY|bank);
+  if (boh == NULLBO) {
+    return false;
+  }
+
+  char *boptr = reinterpret_cast<char *> (xclMapBO(handle, boh, true));
+  if (boptr == nullptr) {
+    free_unmap_bo(handle, boh, boptr, bo_size);
+    return false;
+  }
+
+  memset(boptr, 'A', bo_size);
+
+  if (xclCopyBO(handle, bop2p, boh, bo_size, 0, 0)) {
+    free_unmap_bo(handle, boh, boptr, bo_size);
+    return false;
+  }
+
+  if(xclSyncBO(handle, boh, XCL_BO_SYNC_BO_TO_DEVICE, bo_size, 0)) {
+    free_unmap_bo(handle, boh, boptr, bo_size);
+    return false;
+  }
+
+  if (!p2ptest_set_or_cmp(boptr, bo_size, 'A', false)){
+    return false;
+  }
+
```

Comment:
a comment here would be helpful. Something like:
// testing p2p read flow device -> host